enhancements/ingress/set-replace-append-delete-deny-headers.md

```diff
@@ -0,0 +1,949 @@
+Allowing the cluster-admin or route owner to override the `proxy` HTTP header isn't necessarily a security risk; it would only allow admins or route owners to inject configuration into their own applications, which they already control.  However, the `proxy` header is non-standard, and allowing it to be configured is potentially
+confusing.  For this reason, we do not allow setting the `proxy` header with this EP.
+
+#### Strict-Transport-Security
+Strict-transport-security is already handled separately and we don't want to
+allow overrides of that configuration. If we allowed it, then route owners
+could impact each other's routes. It must be blocked so we have put this
+header in the deny list.
+
+#### X-Forwarded-For
+`x-forwarded-for` and related headers were considered for the deny-list (as modifying these headers using this new API could affect configuration specified using
+another [option](https://docs.openshift.com/container-platform/4.12/networking/ingress-operator.html#nw-using-ingress-forwarded_configuring-ingress))
+but have actual use-cases (which aren't covered by the existing option). They will be allowed, with notes in the API and product documentation,
+and possibly a warning on admission.
+Eg: If someone creates a Ingress Controller with the following:
+```sh
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+name: default
+namespace: openshift-ingress-operator
+spec:
+httpHeaders:
+forwardedHeaderPolicy: Replace
+```
+Then the `set-header` stanzas will be added in `haproxy.config` using the [haproxy-config.template](https://github.com/openshift/router/blob/0cd9a58dff77127562f7851ccfa5cda79b8cef48/images/router/haproxy/conf/haproxy-config.template#L565-L580)
+However, if someone wants to set custom values for these then they need to use the API of this EP.
+If both are set then custom values will take over.
+Please note that the setting of forwardedHeaderPolicy happens at the plain backend section of haproxy.config i.e per route level and not at the front-end section
+of the haproxy.config level as per above mentioned link.
+Another use case is deleting the header which forwardedHeaderPolicy does not provide but this EP does.
+
+#### X-SSL
+`x-ssl` and related headers similarly have an actual use-case, and allowing the cluster admin or route owner to modify these presents no security risk.
+Eg: haproxy takes information from TLS handshake or tls session or the certificate that client provided. These headers are used for mtls and client certificates.
+These headers tell you whether the client provided a certificate, was the certificate valid, what was the content or cipher, distinguished name or expiration date
+on the certificate using converters with prefix like `ssl_` which can be used in the haproxy proxy config to take that information and put it http request headers which get sent
+to the backend application.
```

Comment:
Please check the grammar and spelling again.  Change "a or b or c" to "a, b, or c".  Fix capitalization: "HAProxy", "TLS", "HTTP", "mTLS".  Actually, it would be better to spell out "mTLS" as "mutual TLS".   

On a related note, the IBM Style Guide says to use "for example" instead of "e.g.".

Reply:
done

Reply:
Still not fixed: https://github.com/openshift/enhancements/blob/ae5a9e6428596afcad8ad1c83f801768480c308b/enhancements/ingress/set-delete-http-headers.md#L763